Require Python 3.8. (#715)
glotzerlab/signac: changelog.txt

 - User-provided path functions now raise an error if not unique (#666).
 - ``Collection`` class no longer raises an error when searching by a primary key that does not exist (#676).

Removed
+++++++

>  - Dropped support for Python 3.6 and Python 3.7 (#715). These have been dropped by NEP 29.

Reviewer: ```suggestion
 - Dropped support for Python 3.6 and Python 3.7 (#715) following the recommended support schedules of the [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html).
```
